CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+into commits. As a brief summary, the following golden rules apply
+almost 100% of the time:
+
+- A commit should never break the tests or other CI
+
+  - For example, it should not rely on another commit later in the
+    same PR to fix a bug it introduces.
+
+- A commit should be as small as possible, but no smaller.
+
+  - For example, when adding a new feature, the new tests for that
+    feature should be in the same commit. Ideally the
+    documentation should too, at least if the documentation
+    changes are small.
+
+- A single commit should never mix unrelated changes.
+
+  - For example, refactorings should never be mixed with bugfixes.
+
+The following resources offer useful advice on this topic:
+
+- [GitCommitMessages - OpenStack](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)
+  (short but good; gives examples of what to avoid)
+- [Atomic Commits in the Command Line with Git](https://seesparkbox.com/foundry/atomic_commits_with_git)
+
+## Git commit messages
+
+We aim for high quality commit messages which always have both a title
+and a body, and **always clearly explain the "why"**, i.e. the context
+and motivation for the change, not just the **"how"**.
+
+We adopt the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard as a
+taxonomy for different types of commits, and to track compatibility
+according to [SemVer](https://semver.org/) principles.
+
+The [Perfecting the Patch](https://www.youtube.com/watch?v=pU-VasVPNAs) YouTube video is
+highly recommended, as are the following resources:
+
+- [A guide on commit messages - DEV Community](https://dev.to/yvonnickfrin/a-guide-on-commit-messages-d8n)
+- [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+- [GitCommitMessages - OpenStack](https://wiki.openstack.org/wiki/GitCommitMessages)
+- [https://www.kernel.org/doc/Documentation/SubmittingPatches](https://www.kernel.org/doc/Documentation/SubmittingPatches)
+- [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+- [The Art of the Commit -- A List Apart](https://alistapart.com/article/the-art-of-the-commit)
+- [7 principles for contributing patches to software projects](http://blog.adamspiers.org/2012/11/10/7-principles-for-contributing-patches-to-software-projects/)
+
+## Pull Request - code review
+
+- PRs SHOULD be submitted for each change, in order to ensure that all
+  changes are checked by the CI pipeline.
+
+- For easy review, PRs SHOULD contain:
+
+  - a clear title - in sync with the Liner ticket title
+
+  - a description clearly explaining the **"why"**, i.e. the context. If the issue tracker issue is linked from the PR it's not necessary to duplicate the description, but at least a high level should be given. In addition PR description should document **how** this has been solved, and why this approach was taken.
+
+  - the `CO2-123` identifier for any related issue in the
```

Comment:
Is this applicable for open source contributors?

Reply:
It can be if I opened a GH issue that also has a Linear mirror, like [this one](https://github.com/ToucanProtocol/OffsetHelper/issues/11).